Studio/src/Application/Data/Mesh.cpp

```diff
@@ -117,3 +127,143 @@ vnl_vector<double> Mesh::get_center_transform()
+
+  auto points = this->poly_data_->GetPoints();
+
+  vtkFloatArray* scalars = vtkFloatArray::New();
+  scalars->SetNumberOfValues(points->GetNumberOfPoints());
+  scalars->SetName(name.c_str());
+
+  for (int i = 0; i < points->GetNumberOfPoints(); i++) {
+    double pt[3];
+    points->GetPoint(i, pt);
+
+    ImageType::PointType pitk;
+    pitk[0] = pt[0];
+    pitk[1] = pt[1];
+    pitk[2] = pt[2];
+
+    if (transform.size() == 12) {
+      pitk[0] = pitk[0] + transform[9];
+      pitk[1] = pitk[1] + transform[10];
+      pitk[2] = pitk[2] + transform[11];
+    }
+
+    LinearInterpolatorType::ContinuousIndexType index;
+    image->TransformPhysicalPointToContinuousIndex(pitk, index);
+
+    auto pixel = 0;
+    if (region.IsInside(index)) {
+      pixel = interpolator->EvaluateAtContinuousIndex(index);
+      //  std::cerr << pixel << " ";
+    }
+    scalars->SetValue(i, pixel);
+  }
+  //std::cerr << "\n";
+
+  this->poly_data_->GetPointData()->AddArray(scalars);
+
+  this->poly_data_->GetPointData()->SetScalars(scalars);
+
+  /*
+  auto writer = vtkSmartPointer<vtkPolyDataWriter>::New();
+  writer->SetInputData(this->poly_data_);
+  writer->SetFileName("/tmp/foo.vtk");
+  writer->Update();
+*/
+}
+
+//---------------------------------------------------------------------------
+void Mesh::interpolate_scalars_to_mesh(std::string name, vnl_vector<double> positions,
+                                       itkeigen::VectorXf scalar_values)
+{
+
+  int num_points = positions.size() / 3;
+  if (num_points == 0) {
+    return;
+  }
+
+  vtkSmartPointer<vtkPoints> vtk_points = vtkSmartPointer<vtkPoints>::New();
+
+  vtk_points->SetNumberOfPoints(num_points);
+
+  unsigned int idx = 0;
+  for (int i = 0; i < num_points; i++) {
+
+    double x = positions[idx++];
+    double y = positions[idx++];
+    double z = positions[idx++];
+
+    vtk_points->InsertPoint(i, x, y, z);
+  }
+
+  vtkSmartPointer<vtkPolyData> point_data = vtkSmartPointer<vtkPolyData>::New();
+  point_data->SetPoints(vtk_points);
+
+  vtkSmartPointer<vtkPointLocator> point_locator = vtkPointLocator::New();
+  point_locator->SetDataSet(point_data);
+  point_locator->SetDivisions(100, 100, 100);
+  point_locator->BuildLocator();
+
+  if (!this->poly_data_)
+  {
+    return;
+  }
+  auto points = this->poly_data_->GetPoints();
+
+  vtkFloatArray* scalars = vtkFloatArray::New();
+  scalars->SetNumberOfValues(points->GetNumberOfPoints());
+  scalars->SetName(name.c_str());
+
+  for (int i = 0; i < points->GetNumberOfPoints(); i++) {
+    double pt[3];
+    points->GetPoint(i, pt);
+
+    // find the 8 closest correspondence points the to current point
+    vtkSmartPointer<vtkIdList> closest_points = vtkSmartPointer<vtkIdList>::New();
+    point_locator->FindClosestNPoints(8, pt, closest_points);
+
+    // assign scalar value based on a weighted scheme
+    float weighted_scalar = 0.0f;
+    float distanceSum = 0.0f;
+    float distance[8];
```

Comment:
maybe use `closest_points->GetNumberOfIds()` (or a const int) in case we need to change this? instead of hardcoding 8 in multiple places

Reply:
Yeah, that's a good question.  This is actually just old code from View2 I'm porting over.